include/openmc/source.h

```diff
@@ -43,15 +44,42 @@ extern vector<unique_ptr<Source>> external_sources;
 
 class Source {
 public:
+  // Constructors, destructors
+  Source() = default;
+  explicit Source(pugi::xml_node node);
   virtual ~Source() = default;
 
-  // Methods that must be implemented
-  virtual SourceSite sample(uint64_t* seed) const = 0;
-
   // Methods that can be overridden
-  virtual double strength() const { return 1.0; }
+  virtual SourceSite sample(uint64_t* seed) const;
```

Comment:
It should be possible to make this non-virtual. Let the base classes implement `sample_proposal` as I suggest we rename to, or `sample_no_rejection` as is currently listed.